mars/services/scheduling/supervisor/assigner.py

```diff
@@ -85,11 +85,34 @@ def _get_device_bands(self, is_gpu: bool):
             raise NoMatchingSlots("gpu" if is_gpu else "cpu")
         return filtered_bands
 
-    def _get_random_band(self, is_gpu: bool):
-        avail_bands = self._get_device_bands(is_gpu)
-        return random.choice(avail_bands)
+    def _get_random_band(
+        self,
+        is_gpu: bool,
+        exclude_bands: Set[BandType] = None,
+        exclude_bands_force: bool = False,
```

Comment:
Actually if `exclude_bands_force` is `False`, we return a random which is not empty, so maybe we should keep using `exclude_bands_force` or something else? @wjsi

Reply:
The name `force` itself is not clear enough as what is called `force`. As the behavior actually is that when the flag is on and no bands available, the function should raise instead of returning a random band instead.

Reply:
`allow_empty ` gives me a sense this method will return None if there is no available bands after excluded `exclude_bands` when `allow_empty` is True, but it return a ramdom band actually.